src/modding/ModManager.cs

```diff
@@ -522,30 +763,81 @@ private void UpdateSelectedModInfo()
+                TranslationServer.Translate("THIS_IS_WORKSHOP_MOD") :
+                TranslationServer.Translate("THIS_IS_LOCAL_MOD");
+
+            if (!(string.IsNullOrEmpty(selectedMod.Info.RecommendedThriveVersion) &&
+                    string.IsNullOrEmpty(selectedMod.Info.MinimumThriveVersion)))
+            {
+                selectedModThriveVersionContainer.Visible = true;
+                selectedModRecommendedThriveVersionContainer.Visible =
+                    !string.IsNullOrEmpty(selectedMod.Info.RecommendedThriveVersion);
+                selectedModMinimumThriveVersionContainer.Visible =
+                    !string.IsNullOrEmpty(selectedMod.Info.MinimumThriveVersion);
+                selectedModThriveVersionHSeparator.Visible = selectedModMinimumThriveVersionContainer.Visible &&
+                    selectedModRecommendedThriveVersionContainer.Visible;
+                selectedModRecommendedThriveVersion.Text = selectedMod.Info.RecommendedThriveVersion;
+                selectedModMinimumThriveVersion.Text = selectedMod.Info.MinimumThriveVersion;
+            }
+            else
+            {
+                selectedModThriveVersionContainer.Visible = false;
+            }
+
+            selectedModDescription.ExtendedBbcode = selectedMod.Info.LongDescription ?? selectedMod.Info.Description;
             openModUrlButton.Disabled = selectedMod.Info.InfoUrl == null;
 
-            openModInfoButton.Disabled = false;
+            if (selectedModPreviewImagesContainer.GetChildCount() > 0)
+            {
+                selectedModPreviewImagesContainer.QueueFreeChildren();
+            }
 
-            if (notEnabledMods!.Contains(selectedMod))
+            List<ImageTexture> loadedPreviewImages = LoadModPreviewImages(selectedMod);
+            if (loadedPreviewImages.Count > 0)
             {
-                leftArrow.Disabled = true;
-                rightArrow.Disabled = false;
+                selectedModGalleryContainer.Visible = true;
+                foreach (ImageTexture currentPreviewImage in loadedPreviewImages)
+                {
+                    var currentPreviewImageNode = new TextureRect();
+                    currentPreviewImageNode.Texture = currentPreviewImage;
+                    currentPreviewImageNode.Expand = true;
+                    currentPreviewImageNode.StretchMode = TextureRect.StretchModeEnum.KeepAspectCentered;
+
+                    selectedModPreviewImagesContainer.AddChild(currentPreviewImageNode);
+                }
+
+                selectedModPreviewImagesContainer.CurrentTab = 0;
+                galleryLabel.Text = "1/" + selectedModPreviewImagesContainer.GetTabCount();
+                galleryRightButton.Disabled = selectedModPreviewImagesContainer.CurrentTab >=
+                    selectedModPreviewImagesContainer.GetTabCount() - 1;
+                galleryLeftButton.Disabled = selectedModPreviewImagesContainer.CurrentTab <= 0;
```

Comment:
These two should be in a method of their own so that this logic doesn't need to be duplicated when moving in the gallery.